rust/xaynet-server/src/storage/redis.rs

```diff
@@ -0,0 +1,644 @@
+//! # Redis Data Model
+//!
+//!```text
+//! {
+//!     // Coordinator state
+//!     "coordinator_state": "...", // bincode encoded string
+//!     // Sum dict
+//!     "sum_dict": { // hash
+//!         "SumParticipantPublicKey_1": "SumParticipantEphemeralPublicKey_1",
+//!         "SumParticipantPublicKey_2": "SumParticipantEphemeralPublicKey_2"
+//!     },
+//!     // Seed dict
+//!     "update_participants": [ // set
+//!         "UpdateParticipantPublicKey_1",
+//!         "UpdateParticipantPublicKey_2"
+//!     ],
+//!     "SumParticipantPublicKey_1": { // hash
+//!         "UpdateParticipantPublicKey_1": "EncryptedMaskSeed",
+//!         "UpdateParticipantPublicKey_2": "EncryptedMaskSeed"
+//!     }
+//!     "SumParticipantPublicKey_2": {
+//!         "UpdateParticipantPublicKey_1": "EncryptedMaskSeed",
+//!         "UpdateParticipantPublicKey_2": "EncryptedMaskSeed"
+//!     }
+//!     // Mask dict
+//!     "mask_dict": [ // sorted set
+//!         ("mask_abc", 12341), // (mask: bincode encoded string, score/counter: number)
+//!         ("mask_xyz", 1)
+//!     ]
+//! }
+//! ```
+use crate::{
+    state_machine::coordinator::CoordinatorState,
+    storage::impls::{
+        AddSumParticipant,
+        DeleteSumParticipant,
+        EncryptedMaskSeedRead,
+        EncryptedMaskSeedWrite,
+        MaskObjectRead,
+        MaskObjectWrite,
+        PublicEncryptKeyRead,
+        PublicEncryptKeyWrite,
+        PublicSigningKeyRead,
+        PublicSigningKeyWrite,
+    },
+};
+use redis::{aio::ConnectionManager, AsyncCommands, IntoConnectionInfo, RedisError, RedisResult};
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+use tokio::sync::{OwnedSemaphorePermit, Semaphore};
+use xaynet_core::{
+    mask::{EncryptedMaskSeed, MaskObject},
+    LocalSeedDict,
+    SeedDict,
+    SumDict,
+    SumParticipantEphemeralPublicKey,
+    SumParticipantPublicKey,
+    UpdateParticipantPublicKey,
+};
+
+#[macro_export]
+macro_rules! redis_connection {
```

Comment:
Couldn't this be a method of `Client` instead of a macro?

Reply:
I just tried it out and it is possible. I will remove the macro